client/mysql.cc

```diff
@@ -1116,6 +1116,24 @@ inline int get_command_index(char cmd_char)
   return -1;
 }
 
+static void redact_mysql_pwd(void)
+{
+#ifndef DONT_USE_MYSQL_PWD
+  /* defense-in-depth: overwrite password in original environment vector */
+  char *passwd = getenv("MYSQL_PWD");
+  if (passwd) {
+      /* setenv copies passwd, result not visible in /proc/$pid/environ */
+      if (setenv("MYSQL_PWD", passwd, 1) == -1) {
```

Comment:
By overwriting the password here you've effectively disabled its use entirely (in mysql_real_connect in libmysqld/libmysqld.c).

Reply:
I don't understand your comment. I overwrite the password with itself such that any following `getenv("MYSQL_PWD")` still retrieves the original value. It's just overwritten in the original environment vector.